[5.2] TinyMCE 7

brianteeman opened the pull request:
Pull Request for Issue # .

### Summary of Changes
The majority of changes with 7.x do not impact us as they are for the premium version and the premium plugins. Other changes are really just bug fixes and accessibility improvements.

The full changelog for the open source release of tinymce can be found [here](https://www.tiny.cloud/docs/tinymce/latest/changelog/)

Set tinymce to be used under the gpl licence. https://www.tiny.cloud/docs/tinymce/latest/license-key/#use-tinymce-with-the-gplv2-license - without this change a console warning is displayed

The one change that does need testing is sandbox_iframes

### Testing Instructions
1. Test all editor-xtd buttons 

2. Test the source code editor 
![image](https://github.com/user-attachments/assets/5d0bcba4-d0b5-405d-b663-4b94033e0576)

3. Test any third party plugin eg https://github.com/brianteeman/slashtools
This uses one of the few breaking changes so should fail https://www.tiny.cloud/docs/tinymce/latest/migration-from-6x/#the-autocompleter-ch-configuration-property-has-been-removed-use-the-trigger-property-instead I will make a new release of it but I am not aware of any other plugin using this.

4. Test drag and drop images

5. Test content templates
![image](https://github.com/user-attachments/assets/83e64307-424a-4b32-ab0f-a584f7becedc)

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

brianteeman committed:
[5.2] TinyMCE 7

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
Merge branch '5.2-dev' into tiny7

brianteeman committed:
gpl

Signed-off-by: BrianTeeman <brian@teeman.net>

bembelimen commented:
Hey @brianteeman , not sure if this answers your question, but the idea for the ```sandbox_iframes``` parameter was: when we go to TinyMCE 7 it should still stay at "true", so all entries get sandboxed by default (and the plugin parameter we added to disable it could be removed) but additionally the current domain should be allowed via: [```sandbox-iframes-exclusions```](https://www.tiny.cloud/docs/tinymce/latest/content-filtering/#sandbox-iframes-exclusions).
Also an extra field should be added, where people can add more domains to allow them.

One concern is, how to prevent, that people open the joomla installation with another domain and add iframes for this domain (as it's then allowed automatically).

brianteeman commented:
Thanks, that answers most of it I think. Will add the exclusion field

brianteeman committed:
add new strings and fix the alpha sort

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
add extra field for sandbox exclusions

Signed-off-by: BrianTeeman <brian@teeman.net>

dgrammatiko commented:
@brianteeman @bembelimen according to their docs you shouldn't expose the switch to turn on/off the sandboxing as an empty array is doing exactly that: `To enable [sandboxing](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#sandbox) on all iframes, set this option to an empty array [].`

In sort the `sandbox_iframes_exclusions` should be an array with default value the value of tinymce (is given in the link above): 

```js
[
  'youtube.com',
  'youtu.be',
  'vimeo.com',
  'player.vimeo.com',
  'dailymotion.com',
  'embed.music.apple.com',
  'open.spotify.com',
  'giphy.com',
  'dai.ly',
  'codepen.io'
]
```

brianteeman commented:
yes I know. I am still working on it (or will be tonight when care duties end) or you can submit a pr to this draft branch ;)

brianteeman committed:
wip

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
complete

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
cs

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
cs

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
showon

Signed-off-by: BrianTeeman <brian@teeman.net>

Quy commented:
Should it be only the domain name without `http`?

brianteeman commented:
> Should it be only the domain name without `http`?

??

Quy commented:
These are listed without `http`. Should the additional values be host names or domain names (without `http`)?

```
[
  'youtube.com',
  'youtu.be',
  'vimeo.com',
  'player.vimeo.com',
  'dailymotion.com',
  'embed.music.apple.com',
  'open.spotify.com',
  'giphy.com',
  'dai.ly',
  'codepen.io'
]
```

brianteeman commented:
> These are listed without `http`. Should the additional values be host names or domain names (without `http`)?
> 
> ```
> [
>   'youtube.com',
>   'youtu.be',
>   'vimeo.com',
>   'player.vimeo.com',
>   'dailymotion.com',
>   'embed.music.apple.com',
>   'open.spotify.com',
>   'giphy.com',
>   'dai.ly',
>   'codepen.io'
> ]
> ```

The current domain and this list of common domains are excluded by default.

Quy commented:
Yes, I understand. but when adding new ones to the exclusions list, there is a hint `https://` so maybe it can interpreted to prefix it with `https://`.

Quy commented on plugins/editors/tinymce/src/PluginTraits/DisplayTrait.php:
```diff
@@ -410,6 +410,21 @@ public function prepareTinyMCEUploadPath(string $path): string
+        if (isset($extraOptions->sandbox_iframes_exclusions) && $extraOptions->sandbox_iframes_exclusions) {
+            $exclusionsArray = [];
+            foreach ($extraOptions->sandbox_iframes_exclusions as $value) {
+                if (isset($value->exclusion_url)) {
```
I assume there would be a value at this point so no need to do the check?

brianteeman commented:
I understand you now. Thanks for your perseverance I will look into it in the morning

brianteeman committed:
they are domain names not urls so they should just be type text with no validation thanks @quy

Signed-off-by: BrianTeeman <brian@teeman.net>

Quy commented on administrator/language/en-GB/plg_editors_tinymce.ini:
```diff
@@ -45,25 +45,30 @@ PLG_TINY_FIELD_HTMLWIDTH_LABEL="HTML Width"
 PLG_TINY_FIELD_RESIZE_HORIZONTAL_LABEL="Horizontal Resizing"
 PLG_TINY_FIELD_RESIZING_LABEL="Resizing"
+PLG_TINY_FIELD_SANDBOX_IFRAMES_DESC="This is a security feature that restricts what the iframe can do by setting a sandbox attribute for each iframe. It is recommended to enable this feature for security reasons. Only disable if the iframe is not loading and you're sure that the iframed content is safe."
+PLG_TINY_FIELD_SANDBOX_IFRAMES_EXCLUSIONS_LIST_DESC="Additional URL host names can be added to be excluded from iframe sandboxing. The current domain and this list of common domains are excluded by default. 'youtube.com', 'youtu.be', 'vimeo.com', 'player.vimeo.com', 'dailymotion.com', 'embed.music.apple.com', 'open.spotify.com', 'giphy.com', 'dai.ly', 'codepen.io'."
+PLG_TINY_FIELD_SANDBOX_IFRAMES_EXCLUSIONS_LIST_LABEL="Sandbox Iframe Exclusions List"
+PLG_TINY_FIELD_SANDBOX_IFRAMES_EXCLUSION_DOMAIN_LABEL="URL Host Name"
```
```suggestion
PLG_TINY_FIELD_SANDBOX_IFRAMES_EXCLUSION_DOMAIN_LABEL="Domain Name"
```

Quy replied:
```suggestion
PLG_TINY_FIELD_SANDBOX_IFRAMES_EXCLUSIONS_LIST_DESC="Additional domain names can be added to be excluded from iframe sandboxing. The current domain and this list of common domains are excluded by default. 'youtube.com', 'youtu.be', 'vimeo.com', 'player.vimeo.com', 'dailymotion.com', 'embed.music.apple.com', 'open.spotify.com', 'giphy.com', 'dai.ly', 'codepen.io'."
PLG_TINY_FIELD_SANDBOX_IFRAMES_EXCLUSIONS_LIST_LABEL="Sandbox Iframe Exclusions List"
PLG_TINY_FIELD_SANDBOX_IFRAMES_EXCLUSION_DOMAIN_LABEL="Domain Name"
```

Quy commented on plugins/editors/tinymce/forms/setoptions.xml:
```diff
@@ -238,7 +238,23 @@
+			>
+			<form>
+				<field
+					name="exclusion_url"
```
Rename variable?

Quy commented on plugins/editors/tinymce/src/PluginTraits/DisplayTrait.php:
```diff
@@ -410,6 +410,21 @@ public function prepareTinyMCEUploadPath(string $path): string
         }
 
+        // Add the current domain to the sandbox_iframes_exclusions list
+        $sandboxIframesExclusions = Uri::root();
```
Returns `http://localhost/Joomla_5.2.0-alpha4-dev+pr.43909-Development-Full_Package/`

brianteeman replied:
that is what I would expect. If that is nto what is needed to then I need some help

brianteeman committed:
Update administrator/language/en-GB/plg_editors_tinymce.ini

Co-authored-by: Quy <quy@nomonkeybiz.com>

brianteeman committed:
rename variable

Signed-off-by: BrianTeeman <brian@teeman.net>

brianteeman committed:
Merge branch '5.2-dev' into tiny7

Quy commented on plugins/editors/tinymce/src/PluginTraits/DisplayTrait.php:
```diff
@@ -410,6 +410,21 @@ public function prepareTinyMCEUploadPath(string $path): string
         }
 
+        // Add the current domain to the sandbox_iframes_exclusions list
+        $sandboxIframesExclusions = Uri::root();
```
```suggestion
        $sandboxIframesExclusions = Uri::getInstance()->getHost();
```

brianteeman committed:
Update plugins/editors/tinymce/src/PluginTraits/DisplayTrait.php

Co-authored-by: Quy <quy@nomonkeybiz.com>

brianteeman committed:
Merge branch '5.2-dev' into tiny7

MacJoom committed:
Joomla! 4.4.8 Rc 3

MacJoom committed:
Revert to dev

Signed-off-by: Martin Kopp <kopp05@infotech.ch>

MacJoom committed:
Joomla! 4.4.8 Stable

MacJoom committed:
Revert to dev

Signed-off-by: Martin Kopp <kopp05@infotech.ch>

Fedik committed:
[5.2][Events] Use event classes for Authentication plugins (#43601)

Fedik committed:
[5.2][Events] Use event classes for Finder plugins (#43607)

Hackwar committed:
[5.2] Fixing FQSN in BaseModel (#43995)

muhme committed:
[cypress] PostgreSQL Connection pool for System Tests (#43924)

* Use Connection pool for PostgreSQL in System Tests

Sometimes single specs in the System Tests fail with:
    CypressError: `cy.task('queryDB')` failed with the following error:
    > duplicate key value violates unique constraint "cpostgresmax_users_pkey"

The error happens on different specs in System Tests especially running on drone.
It could be reproduced in running the System Tests tree times (first with installation, 2nd and 3rd run w/o):
    npx cypress run
    npx cypress run --spec 'tests/System/integration/administrator/**/*.cy.{js,jsx,ts,tsx},tests/System/integration/site/**/*.cy.{js,jsx,ts,tsx},tests/System/integration/api/**/*.cy.{js,jsx,ts,tsx},tests/System/integration/plugins/**/*.cy.{js,jsx,ts,tsx}'
    npx cypress run --spec 'tests/System/integration/administrator/**/*.cy.{js,jsx,ts,tsx},tests/System/integration/site/**/*.cy.{js,jsx,ts,tsx},tests/System/integration/api/**/*.cy.{js,jsx,ts,tsx},tests/System/integration/plugins/**/*.cy.{js,jsx,ts,tsx}'

Checking the session usage in pgAdmin shows the maximum number of 100 configured sessions appears to have been reached.
Therefore the usage of postgres connection pool is implemented.
The session usage is reduced to max 12 and the error could no more reproduced.

* Simplify implementation

- Delete function as only called once
- Renamed to postgresConnectionPool

Contributed by @laoneo

---------

Co-authored-by: Allon Moritz <allon.moritz@digital-peak.com>

Fedik committed:
[5.2][Events] Use event classes for Media-action plugins (#43608)

Hackwar committed:
4.4 to 5.2 upmerge 2024-08-29 (#43996)

brianteeman committed:
[5.2] Tinymce lang sort (#43910)

obuisard committed:
[Guided Tours] Fix dropdown mouse accessibility on fancy selects (#43960)

brianteeman committed:
[5.2] Search for content checked out by an author (#43965)

Hackwar committed:
[5.2] SEF: Add option to don't set Itemid to homepage by default (#42989)

* SEF: Add option to don't set Itemid to homepage by default

* codestyle

* Adding option to tags router

* Update Router.php

* Update MenuRules.php

* Switch to strictrouting

* Fixing router behavior for broken URLs

* Fixing bug in system tests

* Removing unnecessary test for user reminder

* Fixing profile edit test to go back to form

Hackwar committed:
Revert "4.4 to 5.2 upmerge 2024-08-29" (#43998)

Hackwar committed:
Merge branch 'refs/heads/4.4-dev' into 44-52-upmerge-20240829

# Conflicts:
#	libraries/src/Version.php

Hackwar committed:
Merge pull request #43999 from Hackwar/44-52-upmerge-20240829

drmenzelit committed:
Articles Module (#43738)

Co-authored-by: Brian Teeman <brian@teeman.net>
Co-authored-by: Martina Scholz <martina@simplysmart-it.de>
Co-authored-by: Benjamin Trenkle <bembelimen@users.noreply.github.com>

obuisard committed:
[5.2] [Guided tours] Auto start tours - full functionality (#43814)

Co-authored-by: Brian Teeman <brian@teeman.net>

brianteeman committed:
Merge branch '5.2-dev' into tiny7

brianteeman commented:
Note to self
Review the skins and themes https://www.tiny.cloud/docs/tinymce/latest/customize-ui/

HLeithner commented:
I rebase this for now to 5.3 (actually 6.0 should be the target), since it's late in the cycle and actually it might be a b/c break if tinymce follows semver. If maintainers/production has finally discussed this a rebase to 5.2 is possible.
So please don't update the branch to 5.3 code

brianteeman commented:
@HLeithner I specifically asked which version to make this for and was told https://github.com/joomla/joomla-cms/issues/43110#issuecomment-2274042689 - please make up your minds
